### PR TITLE
wth - RMID functionalities part 3

### PR DESCRIPTION
**Pull request by wtholt**
Make AJAX request to RMID API to get short_title, long_title and pi_name
from entered RMID record. Fields are disabled upon successful request.
Implemented Gon gem to have access to our environment variables in our
assets. [#136746539]

### CODE REVIEW

File: app/views/protocols/form/_project_roles_fields.html.haml [hunk 1]
```diff
@@ -29,7 +29,7 @@
       .form-group.row
         = pr_form.label :identity_id, t(:protocols)[:authorized_users][:primary_pi], class: 'col-lg-2 control-label required'
         .col-lg-10
-          = pr_form.text_field :identity_id, placeholder: t(:constants)[:prompts][:search_for_user], class: 'form-control'
+          = pr_form.text_field :identity_id, placeholder: t(:constants)[:prompts][:search_for_user], class: 'form-control rm-id-dependent'
```
Comment (kyle-glick) [inline app/views/protocols/form/_project_roles_fields.html.haml]: Needs to be added to the same partial in the dashboard
app/views/dashboard/protocols/form/_project_roles_fields.html.haml

File: app/views/protocols/form/study_form_sections/_study_information.html.haml [hunk 2]
```diff
@@ -30,18 +30,18 @@
       = form.label :research_master_id, class: 'col-lg-2 control-label rm-id' do
         = link_to t(:protocols)[:studies][:information][:research_master_id], RESEARCH_MASTER_LINK
       .col-lg-10
-        = form.text_field :research_master_id, class: 'form-control'
+        = form.text_field :research_master_id, class: 'form-control research-master-field'
```
Comment (kyle-glick) [inline app/views/protocols/form/study_form_sections/_study_information.html.haml]: Ditto
app/views/dashboard/protocols/form/study_form_sections/_study_information.html.haml

File: app/views/protocols/form/_project_roles_fields.html.haml (hunk 1 above)
Comment (kayla-glick) [inline app/views/protocols/form/_project_roles_fields.html.haml:32]: Needs to be added to the same partial in the dashboard
app/views/dashboard/protocols/form/_project_roles_fields.html.haml

File: app/views/protocols/form/study_form_sections/_study_information.html.haml (hunk 2 above)
Comment (kayla-glick) [inline app/views/protocols/form/study_form_sections/_study_information.html.haml:33]: Ditto
app/views/dashboard/protocols/form/study_form_sections/_study_information.html.haml

**Review by kayla-glick — CHANGES_REQUESTED** [review]

**Review by jleonardw9 — APPROVED** [review]

**Review by kayla-glick — APPROVED** [review]

**Pull request merged**